contracts/src/v0.8/upkeeps/GovernanceUpkeep.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+import "../interfaces/KeeperCompatibleInterface.sol";
+import "../vendor/GovernorAlpha.sol";
+
+/// @notice Possible actions that can be taken in the performUpkeep function.
+/// QUEUE => calls 'queue(id)' on the governance contract
+/// EXECUTE => calls 'execute(id)' on the governance contract
+/// CANCEL => calls 'cancel(id)' on the governance contract
+/// UPDATE_INDEX => updates the starting proposal index within the
+///                 upkeep contract to reduce the amount of proposals
+///                 the need to be checked
+enum Action {
+  QUEUE,
+  EXECUTE,
+  CANCEL,
+  UPDATE_INDEX
+}
+
+/// @title Chainlink Keepers Compatible GovernorAlpha Automator
+contract GovernanceAutomator is KeeperCompatibleInterface {
+  GovernorAlpha public immutable s_governanceContract;
+  GovernorAlphaToken public immutable s_governanceTokenContract;
+  uint256 public s_proposalStartingIndex;
+  Action public action;
+
+  constructor(
+    GovernorAlpha _governanceContract,
+    uint256 _proposalStartingIndex,
+    address _tokenContract
```

Comment:
can we make this type `GovernorAlphaToken` instead of type `address` - gives better type safety